contracts/modules/cover/Cover.sol

```diff
@@ -402,6 +416,21 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     emit CoverEdited(coverId, cover.productId, lastCoverSegmentIndex + 1, msg.sender);
   }
 
+  function checkFixedPricingAllowedPools(uint productId, PoolAllocationRequest[] memory allocationRequests) internal {
+
+    uint[] memory allowedPools = fixedPricingAllowedPools[productId];
+    for (uint i = 0; i < allocationRequests.length; i++) {
```

Comment:
the reason i find this loop acceptable is that it's in memory and allowed pools is expected to be small 1,2 items.